atlasdb-api/src/main/java/com/palantir/atlasdb/persistent/api/LogicalPersistentStore.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.persistent.api;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.palantir.atlasdb.persistent.api.PersistentStore.StoreNamespace;
+
+public interface LogicalPersistentStore<K, V> {
+    /**
+     * Retrieve the value associated with the given {@code key}.
+     *
+     * @param storeNamespace from which to retrieve the value
+     * @param key            of the cache entry
+     * @return value associated or null if the entry is missing
+     */
+    @Nullable
```

Comment:
if you are making a new api, can it not be `@Nullable` and use `Optional`'s?

Reply:
It is now optional.